File: src/nodes/increment.zig
const nd = @import("node.zig");

pub const NodeIncrement = struct {
    operand: *nd.Node,

    pub fn init(operand: *nd.Node) NodeIncrement {
        return NodeIncrement{ .operand = operand };
    }

    pub fn get_operand(self: *const NodeIncrement) *nd.Node {
        return self.operand;
    }
};

File: src/nodes/increment.rs
use crate::nodes::node::Node;

pub struct NodeIncrement {
    operand: Node,
}

impl NodeIncrement {
    pub fn init(operand: Node) -> NodeIncrement {
        NodeIncrement { operand }
    }
    
    pub fn get_operand(&self) -> &Node {
        &self.operand
    }
}